n2survey/plot/simple_comparison.py

```diff
@@ -0,0 +1,193 @@
+    y = []
+    for PERCENTAGE in PERCENTAGES:
+        for entry in PERCENTAGE:
+            x.append(entry)
+            y.append(PERCENTAGE[entry])
+    return x, y
+
+
+def form_bar_positions(df, bar_positions=False, totalbar=False, answer_supress=False):
+    """
+    forms a complete list of bar positions for all bars, also the not
+    specified ones.
+    You can specify bar positions in a list, but if you
+    want to specify the last bar you have to specify every bar before that,
+    Maybe this can be improved for more user friendlieness.
+    --> maybe exchange to a bar distance dictionary?
+    if dictionary to complex for standard user, maybe list of tuples?
+    [('Total',1.5),('Yes', 0.2),] etc.
+    with (nameofbar, distance_from_previous_bar)
+    At the moment we calculate PERCENTAGE a second time in this function,
+    but the slowing of the whole plot function is neglectable due to this.
+    """
+    # start list of bar_positions
+    if bar_positions:
+        bar_positions_complete = bar_positions
+    else:
+        bar_positions_complete = [0]
+    COUNT = 0
+    BAR_POS_LENGTH = 0
+    for entry in df:
+        PERCENTAGE = get_percentages(entry, totalbar=totalbar)
+        if answer_supress:
+            for answer in answer_supress:
+                if answer not in PERCENTAGE:
+                    print(f"{answer} does not exist for this question")
+                else:
+                    PERCENTAGE.pop(answer)
+        BAR_POS_LENGTH = BAR_POS_LENGTH + len(PERCENTAGE)
+        totalbar = False
+        if all(
+            [COUNT > 0, len(bar_positions_complete) == BAR_POS_LENGTH - len(PERCENTAGE)]
+        ):
+            bar_positions_complete.append(max(bar_positions_complete) + 1.5)
+        while len(bar_positions_complete) < BAR_POS_LENGTH:
+            bar_positions_complete.append(max(bar_positions_complete) + 1)
+        COUNT = COUNT + 1
+    return bar_positions_complete
+
+
+def simple_comparison_plot(
+    df,
+    answer_supress: Union[list, bool] = False,
+    theme: Optional[Dict] = None,
+    titles: Optional[list] = None,
+    totalbar=False,
+    no_answers=False,
+    hide_titles=False,
+    bar_positions: Union[list, bool] = False,
+    show_percents=True,
+    threshold_percentage=0,
+    bar_width=0.8,
+    legend_columns: int = 2,
+    plot_title: Union[str, bool] = False,
+):
+    """
+    plots correlations from the dataframe df with the existing answer
+    combinations and applies the given specifications.
+    """
+    # form x-axis with answers to first question and y-axis with
+    # percentages of second question correlated to each answer of the first
+    # question.
+    (x, y) = form_x_and_y(df, totalbar=totalbar, answer_supress=answer_supress)
+    # create a list with positonings of the bars in the plot.
+    bar_positions_complete = form_bar_positions(
+        df, bar_positions, totalbar=totalbar, answer_supress=answer_supress
+    )
+    # %% Prepare/Define figure
```

Comment:
What is "%%"?

Reply:
This is a certain feature of Spyder Programming environment. If you add %% it will basically consider the following as a title in the outline so you can easily jump to this part of the script, it helps structuring the code.
%%% then would be a subtitle etc. I found it quite usefull because it helps me to quickly jump to certain parts of the script. If it is not too confusing I would like to leave it in.

Reply:
Interesting. Learned something new!